### PR TITLE
New functions percentage_between() and count_between()

### PR DESCRIPTION
**Pull request by dkuk**
New QL functions described.

### CODE REVIEW

**Comment by dkuk** [comment]
Related pull request - https://github.com/influxdata/influxdb/pull/9946

**Comment by sanderson** [comment]
Thanks @dkuk,

This PR is dependent on the merge of your PR on influxdb. I'll review the content but won't be able to merge it until the PR on influxdb is approved and merged.

If approved, it'll likely be released with InfluxDB 1.6 which means these changes should be made in `content/influxdb/v1.6/query_language/functions.md` (based on the doc's `influxdb-1.6` branch) then submitted as a PR on the `influxdb-1.6` branch.

File: content/influxdb/v1.5/query_language/functions.md [hunk 1]
```diff
@@ -1093,6 +1095,45 @@ tagKey
 location
 ```
 
+## COUNT_BETWEEN()
+Returns number of values that inclusive feets into pointed range of [field value](/influxdb/v1.3/concepts/glossary/#field-value).
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: - "inclusive feets into pointed range" isn't clear.
- `v1.3` in the link needs to be updated to `v1.6`

File: content/influxdb/v1.5/query_language/functions.md [hunk 2]
```diff
@@ -1093,6 +1095,45 @@ tagKey
...
+### Syntax
+```
+SELECT COUNT_BETWEEN(<field_key>,<MIN>,<MAX>)[,<tag_key(s)>|<field_key(s)>] [INTO_clause] FROM_clause [WHERE_clause] [GROUP_BY_clause] [ORDER_BY_clause] [LIMIT_clause] [OFFSET_clause] [SLIMIT_clause] [SOFFSET_clause]
+```
+
+### Description of Syntax
+
+`COUNT_BETWEEN(field_key,MIN,MAX)`
+&emsp;&emsp;&emsp;
+Returns number of fields values which are greater or equal than MIN and less or equal than MAX [field key](/influxdb/v1.3/concepts/glossary/#field-key).
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Needs to be reworded:

> Returns the number of field values greater than or equal to the specified `MIN` field value and less than or equal to the specified `MAX` field value.

If `field value` is going to be a link to the glossary, update `v1.3` to `v1.6`.

File: content/influxdb/v1.5/query_language/functions.md [hunk 3]
```diff
@@ -1093,6 +1095,45 @@ tagKey
...
+
+`COUNT_BETWEEN(/regular_expression/,MIN,MAX)`
+&emsp;&emsp;&emsp;
+Returns number of fields values with each field key that matches the [regular expression](/influxdb/v1.3/query_language/data_exploration/#regular-expressions) which are greater or equal than MIN and less or equal than MAX associated .
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Structure similarly to the above comment.

File: content/influxdb/v1.5/query_language/functions.md [hunk 4]
```diff
@@ -1093,6 +1095,45 @@ tagKey
...
+
+`COUNT_BETWEEN(*,MIN,MAX)`
+&emsp;&emsp;&emsp;
+Returns number of fields values  associated with each field key in the [measurement](/influxdb/v1.3/concepts/glossary/#measurement) which are greater or equal than MIN and less or equal than MAX.
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Structure similarly to the above comment

File: content/influxdb/v1.5/query_language/functions.md [hunk 5]
```diff
@@ -1093,6 +1095,45 @@ tagKey
...
+#### Example 1: Select the number of field values associated with a field key
+```
+> SELECT COUNT_BETWEEN("water_level",0,5) FROM "h2o_feet"
+
+name: h2o_feet
+time                   count_between
+----                   ----------
+2015-08-31T03:42:00Z   9549
+```
+The query returns the number of values that is feets into 0-5ft range (inclusive) in the `water_level` field key and in the `h2o_feet` measurement.
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Needs some adjustment:

> The query returns the number of `water_level` field keys with values between 0 and 5 feet in the `h2o_feet` measurement.

File: content/influxdb/v1.5/query_language/functions.md [hunk 6]
```diff
@@ -1478,6 +1519,44 @@ Notice that the [`GROUP BY time()` clause](/influxdb/v1.3/query_language/data_ex
 The timestamps in the results indicate the the start of each 12-minute time interval;
 the first point in the results covers the time interval between `2015-08-17T23:48:00Z` and just before `2015-08-18T00:00:00Z` and the last point in the results covers the time interval between `2015-08-18T00:24:00Z` and just before `2015-08-18T00:36:00Z`.
 
+## PERCENTAGE_BETWEEN()
+Returns percent of values that inclusive feets into pointed range of [field value](/influxdb/v1.3/concepts/glossary/#field-value).
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: - "inclusive feets into pointed range" isn't clear.
- `v1.3` in the link needs to be updated to `v1.6`

File: content/influxdb/v1.5/query_language/functions.md [hunk 7]
```diff
@@ -1478,6 +1519,44 @@ Notice that the [`GROUP BY time()` clause](/influxdb/v1.3/query_language/data_ex
...
+### Syntax
+```
+SELECT PERCENTAGE_BETWEEN(<field_key>,<MIN>,<MAX>)[,<tag_key(s)>|<field_key(s)>] [INTO_clause] FROM_clause [WHERE_clause] [GROUP_BY_clause] [ORDER_BY_clause] [LIMIT_clause] [OFFSET_clause] [SLIMIT_clause] [SOFFSET_clause]
+```
+
+### Description of Syntax
+
+`PERCENTAGE_BETWEEN(field_key,MIN,MAX)`
+&emsp;&emsp;&emsp;
+Returns percent of fields values which are greater or equal than MIN and less or equal than MAX [field key](/influxdb/v1.3/concepts/glossary/#field-key).
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Use the same recommended structure as above.

File: content/influxdb/v1.5/query_language/functions.md [hunk 8]
```diff
@@ -1478,6 +1519,44 @@ Notice that the [`GROUP BY time()` clause](/influxdb/v1.3/query_language/data_ex
...
+
+`PERCENTAGE_BETWEEN(/regular_expression/,MIN,MAX)`
+&emsp;&emsp;&emsp;
+Returns percent of fields values with each field key that matches the [regular expression](/influxdb/v1.3/query_language/data_exploration/#regular-expressions) which are greater or equal than MIN and less or equal than MAX associated .
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Same as above.

File: content/influxdb/v1.5/query_language/functions.md [hunk 9]
```diff
@@ -1478,6 +1519,44 @@ Notice that the [`GROUP BY time()` clause](/influxdb/v1.3/query_language/data_ex
...
+
+`PERCENTAGE_BETWEEN(*,MIN,MAX)`
+&emsp;&emsp;&emsp;
+Returns percent of fields values  associated with each field key in the [measurement](/influxdb/v1.3/concepts/glossary/#measurement) which are greater or equal than MIN and less or equal than MAX.
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Same as above.

File: content/influxdb/v1.5/query_language/functions.md [hunk 10]
```diff
@@ -1478,6 +1519,44 @@ Notice that the [`GROUP BY time()` clause](/influxdb/v1.3/query_language/data_ex
...
+#### Example 1: Select the percentage of field values associated with a field key
+```
+> SELECT PERCENTAGE_BETWEEN("water_level",0,5) FROM "h2o_feet"
+
+name: h2o_feet
+time                   percentage_between
+----                   ----------
+2015-08-31T03:42:00Z   0.6258356272119544
+```
+The query returns the percent of values that is feets into 0-5ft range (inclusive) in the `water_level` field key and in the `h2o_feet` measurement.
```
Comment (sanderson) [inline content/influxdb/v1.5/query_language/functions.md]: Needs some adjustment:

> The query returns the percent of `water_level` field keys with values between 0 and 5 feet in the `h2o_feet` measurement.

**Comment by dkuk** [comment]
Thank you! Remarks fixed. Close this one please.

New pending request to v1.6 is: #1667

**Pull request closed without merging**